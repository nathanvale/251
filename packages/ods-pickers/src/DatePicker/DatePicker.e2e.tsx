import { E2ETests } from "@origin-digital/ods-types";

export const tests: E2ETests = [];
