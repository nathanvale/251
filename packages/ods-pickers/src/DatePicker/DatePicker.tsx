import * as React from "react";
import { DatePicker as MuiDatePicker } from "@material-ui/pickers";
import {
  DatePickerBaseProps,
  DatePickerBase,
} from "../DatePickerBase/DatePickerBase";

export interface DatePickerProps extends DatePickerBaseProps {}

export const DatePicker = (props: DatePickerProps) => (
  <DatePickerBase {...props} Component={MuiDatePicker} />
);

DatePicker.displayName = "DatePicker";
DatePicker.defaultProps = { "data-id": "date-picker" };
