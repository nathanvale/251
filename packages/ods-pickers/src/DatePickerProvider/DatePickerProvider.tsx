import React from "react";
import DateFnsUtils from "@date-io/date-fns";
import { IUtils } from "@date-io/core/IUtils";
import { MuiPickersUtilsProvider } from "@material-ui/pickers";

export interface DatePickerProviderProps {
  children: React.ReactNode;
  utils?: IUtils<Date>;
}

export const DatePickerProvider = ({
  children,
  utils,
}: DatePickerProviderProps) => (
  <MuiPickersUtilsProvider utils={utils}>{children}</MuiPickersUtilsProvider>
);

DatePickerProvider.defaultProps = {
  utils: DateFnsUtils,
};

DatePickerProvider.displayName = "DatePickerProvider";
