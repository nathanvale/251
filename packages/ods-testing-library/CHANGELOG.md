# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.3.0](https://bitbucket.orgn.io/od/origin-ui/compare/@origin-digital/ods-testing-library@0.2.0...@origin-digital/ods-testing-library@0.3.0) (2020-02-12)


### Features

* **tests, columns:** alignY default changed to top ([c23c964](https://bitbucket.orgn.io/od/origin-ui/commits/c23c964))





# 0.2.0 (2020-01-31)


### Features

* set up jest for snapshots ([6e6fbfa](https://bitbucket.orgn.io/od/origin-ui/commits/6e6fbfa))
