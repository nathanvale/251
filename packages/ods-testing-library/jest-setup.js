import { configure } from "@testing-library/react";

require("jest-styled-components");

configure({ testIdAttribute: "data-id" });
