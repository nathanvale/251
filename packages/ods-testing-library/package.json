{
  "name": "@origin-digital/ods-testing-library",
  "description": "ODS testing tools that enhance react-testing-library",
  "license": "UNLICENSED",
  "author": "Nathan Vale <nathan.vale@origin.com.au>",
  "version": "0.5.0",
  "main": "dist/cjs/index.js",
  "module": "dist/es/index.js",
  "files": [
    "dist",
    "jest-setup.js"
  ],
  "scripts": {
    "build": "yarn build:cjs && yarn build:es",
    "build:cjs": "BUILD_FORMAT=cjs origin-scripts build --monorepo --delete-dir-on-start --out-dir dist/cjs",
    "build:es": "BUILD_WEBPACK=true origin-scripts build --monorepo --out-dir dist/es"
  },
  "sideEffects": false,
  "typings": "dist/typings/index.d.ts",
  "dependencies": {
    "@babel/runtime-corejs3": "^7.8.3",
    "@testing-library/jest-dom": "^5.1.1"
  },
  "peerDependencies": {
    "@origin-digital/ods-themes": "*",
    "@testing-library/react": "9.x",
    "jest-styled-components": "6.x",
    "react": "^16.9.0"
  },
  "devDependencies": {
    "@testing-library/react": "9.x"
  },
  "publishConfig": {
    "registry": "https://artifactory.origindigital-dac.com.au/artifactory/api/npm/npm-internal/"
  }
}
