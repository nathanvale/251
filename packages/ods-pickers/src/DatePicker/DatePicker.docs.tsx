import React from "react";
import { MaterialUiPickersDate } from "@material-ui/pickers/typings/date";
import { ComponentDocs } from "@origin-digital/ods-types";
import { isWeekend, isBefore, addDays } from "date-fns";
import { DatePickerProvider } from "../DatePickerProvider/DatePickerProvider";
import { propDescriptions } from "../DatePickerBase/DatePickerBase";
import { DatePicker, DatePickerProps } from "./DatePicker";

export const docs: ComponentDocs<DatePickerProps> = {
  category: "Interaction",
  componentName: "DatePicker",
  description:
    "A DatePicker is a special type of input field, which renders a calendar from which the user can select a date. For improved accessibility via keyboard date entry, please use KeyboardDatePicker if possible. Note: Your app should be wrapped in a DatePickerProvider for this to work.",
  propDescriptions,
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <DatePickerProvider>
          <DatePicker id="datepicker" defaultValue={new Date()} />
        </DatePickerProvider>
      ),
    },
    additional: [
      {
        label: "Label",
        Code: () => (
          <DatePickerProvider>
            <DatePicker id="datepicker" label="Date of birth" disableFuture />
          </DatePickerProvider>
        ),
      },
      {
        label: "Format",
        Code: () => (
          <DatePickerProvider>
            <DatePicker
              id="datepicker"
              label="Date of birth"
              defaultValue={new Date(2020, 1, 1)}
              format="d/M/yyyy"
            />
          </DatePickerProvider>
        ),
      },
      {
        label: "Disabled picker",
        Code: () => (
          <DatePickerProvider>
            <DatePicker id="datepicker" label="Connection date" disabled />
          </DatePickerProvider>
        ),
      },
      {
        label: "Controlled",
        codeString: `
        const [selectedDate, handleDateChange] = React.useState(new Date());
        return (
          <DatePickerProvider>
            <DatePicker
              id="datepicker"
              label="Date of birth"
              value={selectedDate}
              onChange={handleDateChange}
            />
          </DatePickerProvider>
        );
        `,
        Code: () => {
          const [selectedDate, handleDateChange] = React.useState<
            MaterialUiPickersDate
          >(new Date());
          return (
            <DatePickerProvider>
              <DatePicker
                id="datepicker"
                label="Date of birth"
                value={selectedDate}
                onChange={handleDateChange}
              />
            </DatePickerProvider>
          );
        },
        playroom: false,
      },
      {
        label: "Disable dates",
        description: "In this example we disable dates in the picker",
        codeString: `
        import { isWeekend, isBefore, addDays } from "date-fns";
        import { DatePickerProvider } from "../DatePickerProvider/DatePickerProvider";
        import { DatePicker, DatePickerProps, defaultDateFormat } from "./DatePicker";

        const isUnavailableDate = (date: MaterialUiPickersDate) => {
          if (!date) {
            return false;
          }
          return isWeekend(date) || isBefore(date, addDays(new Date(), 1));
        };
        const [selectedDate, setSelectedDate] = React.useState<
          MaterialUiPickersDate
        >();
        const handleChange = (date: MaterialUiPickersDate) =>
          setSelectedDate(date);

        return (
          <DatePickerProvider>
            <DatePicker
              id="datepicker"
              label="Connection date (DD/MM/YYYY)"
              value={selectedDate}
              onChange={handleChange}
              shouldDisableDate={isUnavailableDate}
            />
          </DatePickerProvider>
        );
      `,
        Code: () => {
          const isUnavailableDate = (date: MaterialUiPickersDate) => {
            if (!date) {
              return false;
            }
            return isWeekend(date) || isBefore(date, addDays(new Date(), 1));
          };
          const [selectedDate, setSelectedDate] = React.useState<
            MaterialUiPickersDate
          >();
          const handleChange = (date: MaterialUiPickersDate) =>
            setSelectedDate(date);

          return (
            <DatePickerProvider>
              <DatePicker
                id="datepicker"
                label="Connection date (DD/MM/YYYY)"
                value={selectedDate}
                onChange={handleChange}
                shouldDisableDate={isUnavailableDate}
              />
            </DatePickerProvider>
          );
        },
        playroom: false,
      },
    ],
  },
  snippets: [
    {
      label: "Simple case",
      Code: () => (
        <DatePickerProvider>
          <DatePicker id="datepicker" defaultValue={new Date()} />
        </DatePickerProvider>
      ),
    },
    {
      label: "Disable future dates",
      Code: () => (
        <DatePickerProvider>
          <DatePicker id="datepicker" label="Date of birth" disableFuture />
        </DatePickerProvider>
      ),
    },
    {
      label: "Disable past dates",
      Code: () => (
        <DatePickerProvider>
          <DatePicker id="datepicker" label="Connection date" disablePast />
        </DatePickerProvider>
      ),
    },
    {
      label: "Custom format",
      Code: () => (
        <DatePickerProvider>
          <DatePicker
            id="datepicker"
            label="Connection date"
            format="d/M/yyyy"
          />
        </DatePickerProvider>
      ),
    },
  ],
};
