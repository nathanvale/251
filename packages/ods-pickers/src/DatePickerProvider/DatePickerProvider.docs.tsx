import React from "react";
import { ComponentDocs } from "@origin-digital/ods-types";
import { DatePicker } from "../DatePicker/DatePicker";
import {
  DatePickerProvider,
  DatePickerProviderProps,
} from "./DatePickerProvider";

export const docs: ComponentDocs<DatePickerProviderProps> = {
  category: "Logic",
  componentName: "DatePickerProvider",
  description:
    "DatePickerProvider should wrap your app and allows you to pass in a date management library of your choice. If one is not provided, it will default to date-fns. Under the hood this is just a wrapper for MuiPickersUtilsProvider https://material-ui-pickers.dev/getting-started/usage",
  propDescriptions: { utils: "Date management lib of your choice" },
  migrationGuide: false,
  examples: {
    default: {
      Code: () => (
        <DatePickerProvider>
          <DatePicker
            id="datepicker"
            label="Connection date"
            defaultValue={new Date()}
          />
        </DatePickerProvider>
      ),
    },
  },
  snippets: [],
};
