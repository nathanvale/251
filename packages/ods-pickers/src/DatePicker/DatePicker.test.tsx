import React from "react";
import { render } from "@origin-digital/ods-testing-library";
import { DatePickerProvider } from "../DatePickerProvider/DatePickerProvider";
import { DatePicker } from "./DatePicker";

test("it renders", () => {
  const { container } = render(
    <DatePickerProvider>
      <DatePicker
        id="datepicker"
        data-id="test"
        label="Date of birth"
        value={new Date(1977, 5, 8)}
        helperText="Enter your date of birth"
      />
    </DatePickerProvider>
  );
  expect(container).toMatchSnapshot();
});

test("it should render error", () => {
  const { container } = render(
    <DatePickerProvider>
      <DatePicker id="datepicker" error />
    </DatePickerProvider>
  );
  expect(container).toMatchSnapshot();
});
