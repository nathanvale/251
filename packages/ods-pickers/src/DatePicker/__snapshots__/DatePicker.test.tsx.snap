// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-2 MuiFormControl-fullWidth"
    data-id="test"
    format="dd/MM/yyyy"
  >
    <label
      class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-filled MuiFormLabel-filled"
      data-id="test-label"
      data-shrink="true"
      for="datepicker"
      id="datepicker-label"
    >
      Date of birth
    </label>
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-describedby="datepicker-helper-text"
        aria-invalid="false"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input"
        data-id="test-input"
        id="datepicker"
        readonly=""
        type="text"
        value="08/06/1977"
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained form-helper-text-grey400-colour-3 MuiFormHelperText-filled"
      data-id="test-helper-text"
      id="datepicker-helper-text"
    >
      Enter your date of birth
    </p>
  </div>
</div>
`;

exports[`it should render error 1`] = `
<div>
  <div
    class="MuiFormControl-root MuiTextField-root textfieldbase-root-5 MuiFormControl-fullWidth"
    data-id="date-picker"
    format="dd/MM/yyyy"
  >
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiFilledInput-underline Mui-error Mui-error MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-describedby="datepicker-helper-text"
        aria-invalid="true"
        autocomplete="false"
        class="MuiInputBase-input MuiFilledInput-input"
        data-id="date-picker-input"
        id="datepicker"
        readonly=""
        type="text"
        value=""
      />
    </div>
    <p
      class="MuiFormHelperText-root MuiFormHelperText-contained form-helper-text-grey400-colour-6 Mui-error"
      data-id="date-picker-helper-text"
      id="datepicker-helper-text"
    >
      <div>
        Â 
      </div>
    </p>
  </div>
</div>
`;
